src/reducers/reducer.js

```diff
@@ -1411,6 +1413,34 @@ export const onNanoContractHistoryFailure = (state, { payload }) => ({
   },
 });
 
+/**
+ * @param {Object} state
+ * @param {{
+ *   payload: {
+ *     ncId: string;
+ *   }
+ * }} action
+ */
+export const onNanoContractHistoryClean = (state, { payload }) => ({
+  ...state,
+  nanoContract: {
+    ...state.nanoContract,
+    history: {
+      ...state.nanoContract.history,
+      [payload.ncId]: [],
+    },
+    historyMeta: {
+      ...state.nanoContract.historyMeta,
+      [payload.ncId]: {
+        ...(state.nanoContract.historyMeta[payload.ncId]),
```

Comment:
Why do you need this if you are setting the default for all attributes of `historyMeta`?

Reply:
Just to be future proof. We don't know which properties we can add to historyMeta upfront, then if it grows we don't break anything. But this is just a way of thinking, we you prefer I can remove it.